lib/Cake/Event/CakeEventManager.php

```diff
@@ -166,7 +166,8 @@ protected function _extractCallable($function, $object) {
  */
 	public function detach($callable, $eventKey = null) {
 		if ($callable instanceof CakeEventListener) {
-			return $this->_detachSubscriber($callable, $eventKey);
+			$this->_detachSubscriber($callable, $eventKey);
+			return;
```

Comment:
I think this change also should be reverted. Currently, this method allows people to return any types by overriding `_detachSubscriber()`.

Reply:
Is Event::detach() being checked somewhere? Because this method here also should return void and not be checked for return values ideally.

Reply:
Ideally, you are right. But people might write such code. For example, talking about Table::find() in 3.x, you could override it to return `int` to support `find('count')`, despite of that the phpDoc says it `@return \Cake\ORM\Query`.

Reply:
I got it. I got smarter. reverted code.